tools/SkeletonExtraction/CGAL/CreateSkeleton2.cpp

```diff
@@ -2,78 +2,237 @@
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Filtered_predicate.h>
+#include <CGAL/MP_Float.h>
+#include <CGAL/Cartesian_converter.h>
+#include <CGAL/convex_hull_2.h>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+typedef K::Point_2                   Point;
 typedef CGAL::Polygon_2<K>           Polygon_2;
-typedef CGAL::Straight_skeleton_2<K> Ss ;
-typedef boost::shared_ptr<Ss> SsPtr ;
-
-using namespace PyMesh;
+typedef CGAL::Straight_skeleton_2<K> Ss;
+typedef boost::shared_ptr<Ss> SsPtr;
 
-void CreateSkeleton2::run(const MatrixFr& points)
+namespace PyMesh
 {
-    Polygon_2 poly;
-    std::size_t num_vertices = points.rows();
-
-    for (size_t i=0; i<num_vertices; i++) {
-        const VectorF& v = points.row(i);
-        poly.push_back( Point(v[0],v[1]) );
+    template<class K>
+    void print_point ( CGAL::Point_2<K> const& p )
+    {
+        std::cout << "(" << p.x() << "," << p.y() << ")" << std::endl;
     }
 
-    std::vector<Point>                          skeleton_points;   //skeliton points
-    std::set<std::set<std::size_t>>             skeleton_edges;    //skeleton edges represented as point indices 
+    void compute_skeleton(const MatrixFr& points,
+                          const bool direction,
+                          Matrix2Fr& computed_vertices, 
+                          Matrix2Ir& computed_edges )
+    {
+        std::cout << "compute_skeleton start" << std::endl;
+        std::size_t num_vertices = points.rows();
+
+        Polygon_2 poly;
+        if( !direction ){
+            for (size_t i=0; i<num_vertices; i++) {
+                const VectorF& v = points.row(i);
+                Point point(v[0],v[1]);
+                poly.push_back( std::move(point) );
+            }
+        }
+        else
+        {
+            for (int i=num_vertices-1; i>=0; i--) {
+                const VectorF& v = points.row(i);
+                Point point(v[0],v[1]);
+                poly.push_back( std::move(point) );
+            }
+        }
 
-    //compute skeleton
-    SsPtr ss = CGAL::create_interior_straight_skeleton_2(poly.vertices_begin(), poly.vertices_end());
+        std::vector<Point>                          skeleton_points;   //skeliton points
+        std::set<std::set<std::size_t>>             skeleton_edges;    //skeleton edges represented as point indices
 
-    std::map<size_t,Point> skeleton_map_points;//tmp object
-    typedef typename Ss::Halfedge_const_iterator Halfedge_const_iterator;
-    for ( Halfedge_const_iterator i = ss->halfedges_begin(); i != ss->halfedges_end(); ++i )
-    {
-        if( i->is_bisector() )
+        //compute skeleton
+        SsPtr ss = CGAL::create_interior_straight_skeleton_2(poly);
+        std::map<size_t,Point> skeleton_map_points;//tmp object
+        typedef typename Ss::Halfedge_const_iterator Halfedge_const_iterator;
+        for ( Halfedge_const_iterator i = ss->halfedges_begin(); i != ss->halfedges_end(); ++i )
```

Comment:
Maybe use `for(auto)` syntax, it would be easier to read and not require the typedef above